sound/soc/intel/boards/sof_da7219_max98373.c

```diff
@@ -146,17 +156,24 @@ static int da7219_codec_init(struct snd_soc_pcm_runtime *rtd)
 
 static int speaker_amp_init(struct snd_soc_pcm_runtime *rtd)
 {
-	int ret;
+	const struct snd_soc_dapm_widget *snd_widget = max98373_widgets;
+	const struct snd_soc_dapm_route *snd_route = max98373_map;
+	int ret, size = 2;
+
+	/* Note card name has sof- prefix */
+	if (!strcmp(rtd->card->name, "sof-da7219max98360a")) {
+		snd_widget = max98360a_widgets;
+		snd_route = max98360a_map;
+		size = 1;
+	}
 
 	/* Add widgets */
-	ret = snd_soc_dapm_new_controls(&rtd->card->dapm, max98373_widgets,
-					ARRAY_SIZE(max98373_widgets));
+	ret = snd_soc_dapm_new_controls(&rtd->card->dapm, snd_widget, size);
 	if (ret)
 		return ret;
 
 	/* Add routes */
-	return snd_soc_dapm_add_routes(&rtd->card->dapm, max98373_map,
-				       ARRAY_SIZE(max98373_map));
+	return snd_soc_dapm_add_routes(&rtd->card->dapm, snd_route, size);
```

Comment:
well if you go dynamic you need to remove the status parts:

````c
static struct snd_soc_card card_da7219_m98360a = {
	.name = "da7219max98360a",
	.owner = THIS_MODULE,
	.dai_link = dais,
	.num_links = ARRAY_SIZE(dais),
	.controls = m98360a_controls,
	.num_controls = ARRAY_SIZE(m98360a_controls),
	.dapm_widgets = widgets,
	.num_dapm_widgets = ARRAY_SIZE(widgets),
	.dapm_routes = audio_map,
	.num_dapm_routes = ARRAY_SIZE(audio_map),

static struct snd_soc_card card_da7219_m98373 = {
	.name = "da7219max",
	.owner = THIS_MODULE,
	.dai_link = dais,
	.num_links = ARRAY_SIZE(dais),
	.controls = controls,
	.num_controls = ARRAY_SIZE(controls),
	.dapm_widgets = widgets,
	.num_dapm_widgets = ARRAY_SIZE(widgets),
	.dapm_routes = audio_map,
	.num_dapm_routes = ARRAY_SIZE(audio_map),
````
doing the same thing in different places isn't so good in general.

Reply:
Yes, thanking for pointing that :) will fix in next update.